File: server/java/src/main/java/RDS/querys/node_query.java
package RDS.querys;

import RDS.models.Node;

public class node_query {
    static public String addNode(Node n){
       return "INSERT INTO \"public\".\"rs_nodes\" (\"osm_Id\", \"node_Id\", \"latitude\", \"longitude\", \"degree\", \"edges\", \"tags\") " +
                "VALUES " +
                "('"+n.getOsm_Id()+"','"+ n.getNode_id() +"', "+n.getLatitude()+", "+n.getLongitude()+", "+n.getDegree()+", '"+n.getEdges()+"', '"+n.getTags()+"');";
    }
    static public String updateNode(Node n) {
        return "UPDATE \"public\".\"rs_nodes\" SET " +
                "\"latitude\" = "+n.getLatitude()+", " +
                "\"longitude\" = "+n.getLongitude()+"," +
                "\"degree\" = "+n.getDegree()+", " +
                "\"edges\" = '"+n.getEdges()+"', " +
                "\"tags\" = '"+n.getTags()+"' " +
                "WHERE \"node_Id\" = '"+n.getOsm_Id()+"';";
    }
    static public String deleteNodeById(Node n) {
        return "DELETE FROM \"public\".\"rs_nodes\" WHERE \"node_Id\" = '"+n.getOsm_Id()+"';";
    }
    static public String getNodeById(Node n) {
        return "SELECT * FROM \"public\".\"rs_nodes\" WHERE \"node_Id\" = '"+n.getOsm_Id()+"';";
    }
    static public String getAllNodes() {
        return "SELECT * FROM \"public\".\"rs_nodes\";";
    }
}

File: server/node/RDS/querys/node_query.js
const node_query = {
    addNode: (node) => {
        return `INSERT INTO "public"."rs_nodes" 
        ("osm_Id", "node_Id", "latitude", "longitude", "degree", "edges", "tags") 
        VALUES 
        (${node.osm_Id}, '${node.node_Id}', ${node.latitude}, ${node.longitude}, ${node.degree}, '${node.edges}', '${node.tags}');`;
    },
    updateNode: (node) => {
        return `UPDATE "public"."rs_nodes" SET 
                    "osm_Id" = ${node.osm_Id}, 
                    "latitude" = ${node.latitude}, 
                    "longitude" = ${node.longitude},
                    "degree" = ${node.degree}, 
                    "edges" = '${node.edges}', 
                    "tags" = '${node.tags}'
                WHERE "node_Id" = '${node.node_Id}';`;
    },
    deleteNodeById: (node) => {
        return `DELETE FROM "public"."rs_nodes" 
                WHERE "node_Id" = '${node.node_Id}';`;
    },
    getNodeById: (node) => {
        return `SELECT * FROM "public"."rs_nodes" 
        WHERE "node_Id" = '${node.node_Id}';`;
    },
    getAllNodes: (node) => {
        return `SELECT * FROM "public"."rs_nodes";`;
    },
}

export default node_query;
